### ISSUE TITLE
ArangoDB crashes from time to time

### ISSUE THREAD

**Issue by ra0o0f** [arangodb/arangodb#1615]
Hi. in my development machine ArangoDB server crashes once for every 3 or 4 days, i tried to turn on log on `debug` and here what i got:

os windows 8 64bit
ArangoDB 2.7

log file size is 94mb, these are last entries
```
2015-12-16T23:14:38 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: (function (params) { (function () {\n      require('org/arangodb/foxx/queues/manager').manage();\n    })(params) } )(params);
2015-12-16T23:14:39 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: (function (params) { (function () {\n      require('org/arangodb/foxx/queues/manager').manage();\n    })(params) } )(params);
2015-12-16T23:14:39 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: HttpServerJob
2015-12-16T23:14:40 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: (function (params) { (function () {\n      require('org/arangodb/foxx/queues/manager').manage();\n    })(params) } )(params);
2015-12-16T23:14:40 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\VocBase\datafile.cpp:842] reached end of datafile 'C:\Program Files\ArangoDB 2.7.0\bin\..\var\lib\arangodb\databases\database-76067\collection-2500899\journal-2894115.db' data, current size 1688
2015-12-16T23:14:41 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: (function (params) { require('org/arangodb/statistics').historian(); } )(params);
2015-12-16T23:14:41 [1300] DEBUG [C:\b\ArangoDB-2.7.0\lib\Basics\Thread.cpp:304] set asynchronous cancelation for thread 'dispat_std'
2015-12-16T23:14:41 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: (function (params) { (function () {\n      require('org/arangodb/foxx/queues/manager').manage();\n    })(params) } )(params);
2015-12-16T23:14:41 [1300] ERROR Unhandled exception: -1073741819 - will crash now.
2015-12-16T23:14:42 [1300] DEBUG [C:\b\ArangoDB-2.7.0\arangod\Dispatcher\DispatcherThread.cpp:181] starting to run job: (function (params) { (function () {\n      require('org/arangodb/foxx/queues/manager').manage();\n    })(params) } )(params);

```

**Comment by jsteemann** [comment]
Note to self: exception code `-1073741819` is the hex representation of the 4 byte value 0xc0000005, which on Windows means "Access Violation".

**Comment by dothebart** [comment]
Recommendation No1: please upgrade to the latest 2.7.2 since we may have fixes in there which may fix your issue.
To get further informations about incidents happening, we would need a minidump of such a hanging process. 

On the procedure of aquiring such a minidump:
Open the windows task-manager, change its view to 'More details', identify the hanging arangod process. open the context of the task manager over the line with the 'arangod' or 'arangodb' and select 'Create Dump File'
Once that is finished a Dialog appears with the URL to the file created. Its important to keep the dialog open until you locate it in explorer, since its a hidden folder you need to enter the URL in the location bar by hand.

More information about automatically generating dumps from crashes using procump can be found here: https://github.com/arangodb/arangodb/blob/devel/README_maintainers.md#windows-debugging

Dump files may be huge, so please compress it (i.e. with 7zip) and upload it somewhere ala dropbox so we can download it.

Then we may analyze the dump file and find the problem.

**Comment by ra0o0f** [comment]
thanks @dothebart, it's not a hanging process, arangod will stop, can i still create dump in this state?
and crashing is happen irregularly(once for every 3 or 4 day)

**Comment by dothebart** [comment]
you need to have `procdump` running attached to the arangod.exe in order to get these crash dumps; Please see the github link above for details on where to find and howto set up.

**Comment by dothebart** [comment]
Have you been able to reproducy this? So far it seems as if 2.8.1 has improved the situation. If its not reproducable, can we close this for now?

**Comment by ra0o0f** [comment]
@dothebart it happened 5 or 6 times since this issue is opened, sorry if i didn't inform you here , it occurred very randomly. close this if you want, i will upgrade to 2.8 and if it happened i will open a new issue with process dump details.

**Comment by dothebart** [comment]
okies.
